clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3872,6 +3872,13 @@ LangAS CodeGenModule::GetGlobalVarAddressSpace(const VarDecl *D) {
     if (!D || D->getType().getAddressSpace() == LangAS::Default) {
       return LangAS::opencl_global;
     }
+    if (D)
```

Comment:
What cases exist where the VarDecl is null?

Reply:
I don't know. Following the above `if (!D ||...)` style

Reply:
this one is removed entirely